corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/entities/app_selection.js

```diff
@@ -0,0 +1,62 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("Entities", function (Entities, FormplayerFrontend, Backbone, Marionette, $) {
+    Entities.AppSelect = Backbone.Model.extend({
+        urlRoot: "appSelects",
+    });
+
+    Entities.configureStorage("FormplayerFrontend.Entities.AppSelect");
+
+    Entities.AppSelectCollection = Backbone.Collection.extend({
+        url: "appSelects",
+        model: Entities.AppSelect,
+    });
+
+    Entities.configureStorage("FormplayerFrontend.Entities.AppSelectCollection");
+
+    var storeApps = function (apps) {
+        var old_apps = new Entities.AppSelectCollection();
+        var defer = $.Deferred();
+        old_apps.fetch({
+            success: function (data) {
+                defer.resolve(data);
+            },
+        });
+        var promise = defer.promise();
+        $.when(promise).done(function (oldApps) {
+            // clear app's local storage when we load new list of apps
+            window.localStorage.clear();
+            apps = new Entities.AppSelectCollection(apps);
+            apps.forEach(function (app) {
+                app.save();
+            });
+            oldApps.reset(apps.models);
```

Comment:
you might be able to use `reset: true` as a fetch parameter and get rid of some of this logic